src/awkward/operations/ak_without_field.py

```diff
@@ -0,0 +1,93 @@
+# BSD 3-Clause License; see https://github.com/scikit-hep/awkward-1.0/blob/main/LICENSE
+
+from collections.abc import Sequence
+
+import awkward as ak
+
+np = ak._nplikes.NumpyMetadata.instance()
+
+
+def without_field(array, where, *, highlevel=True, behavior=None):
+    """
+    Args:
+        array: Array-like data (anything #ak.to_layout recognizes).
+        where (str or non-empy sequence of str): If str, the name of the field
+            to be removed. If a sequence, it is interpreted as a path where to
+            remove the field in a nested record.
+        highlevel (bool): If True, return an #ak.Array; otherwise, return
+            a low-level #ak.contents.Content subclass.
+        behavior (None or dict): Custom #ak.behavior for the output array, if
+            high-level.
+
+    Returns an #ak.Array or #ak.Record (or low-level equivalent, if
+    `highlevel=False`) with an existing field removed. This function does not
+    change the array in-place.
+
+    See #ak.Array.__delitem__ and #ak.Record.__delitem__ for a variant that
+    changes the high-level object in-place. (These methods internally use
+    #ak.without_field, so performance is not a factor in choosing one over the
+    other.)
+    """
+    with ak._errors.OperationErrorContext(
+        "ak.without_field",
+        dict(array=array, where=where, highlevel=highlevel, behavior=behavior),
+    ):
+        return _impl(array, where, highlevel, behavior)
+
+
+def _impl(base, where, highlevel, behavior):
+    if isinstance(where, str):
+        where = [where]
+    elif not (isinstance(where, Sequence) and all(isinstance(x, str) for x in where)):
+        raise ak._errors.wrap_error(
+            TypeError(
+                "Field names must be given as a single string, or a sequence of strings"
+            )
+        )
+
+    behavior = ak._util.behavior_of(base, behavior=behavior)
+    base = ak.operations.to_layout(base, allow_record=True, allow_other=False)
+
+    def action(layout, lateral_context, **kwargs):
+        if isinstance(layout, ak.contents.RecordArray):
+            field, *next_where = lateral_context["where"]
```

Comment:
The conceptual difference is the pattern of updates:

https://awkward-array.org/doc/main/reference/generated/ak.transform.html#contexts

If a subtree is not supposed to communicate with any other subtree, as is the case here, `depth_context` is semantically correct. `lateral_context` is for if you _want_ it to accumulate as in the example linked above.

I guess it works for either context type here, since you're copying dicts in the implementation. I can see that if you switch to `depth_context`, there would be an additional (unnecessary) copy, but it's a dict of one field.

So yeah, please do switch to `depth_context`.